charts/airbyte-bootloader/templates/pod.yaml

```diff
@@ -71,8 +70,43 @@ spec:
               name: {{ .Values.global.configMapName | default (printf "%s-airbyte-secrets" .Release.Name) }}
               key: DATABASE_USER
         {{- end }}
+        # Values from secret
+        {{- if .Values.secrets }}
+        {{- range $k, $v := .Values.secrets }}
+        - name: {{ $k }}
+          valueFrom:
+            secretKeyRef:
+              name: bootloader-secrets
+              key: {{ $k }}
+        {{- end }}
+        {{- end }}
+
+        # Values from env
         {{- if .Values.extraEnv }}
-        {{ .Values.extraEnv | toYaml | nindent 10 }}
+        {{- range $k, $v := mergeOverwrite .Values.extraEnv .Values.global.env_vars }}
+        - name: {{ $k }}
+          value: {{ $v | quote }}
+        {{- end }}
+        {{- end }}
+
+        {{- if and .Values.gsm .Values.global.env }}
+        {{- range $k, $v := mergeOverwrite .Values.gsm .Values.global.gsm }}
+        - name: {{ $k }}
+          valueFrom:
+            secretKeyRef:
+              name: {{ printf "%s-%s" $.Values.global.env $v }}
+              key: {{ $k }}
+            {{- end }}
+        {{- end }}
+
+        {{- if and .Values.gsm (not .Values.global.env) }}
```

Comment:
we should not leak the gsm abstraction out onto OSS charts.

Reply:
Removed from oss, logic will be handeled by another approach